tools/rtpsrelay/HandlerStatisticsListener.cpp

```diff
@@ -0,0 +1,56 @@
+#include "HandlerStatisticsListener.h"
+
+#include "utility.h"
+
+#include "lib/RelayTypeSupportImpl.h"
+#include "lib/QosIndex.h"
+
+namespace RtpsRelay {
+
+void HandlerStatisticsListener::on_data_available(DDS::DataReader_ptr reader)
+{
+  HandlerStatisticsDataReader_var dr = HandlerStatisticsDataReader::_narrow(reader);
+  if (!dr) {
+    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) %N:%l ERROR: HandlerStatisticsListener::on_data_available failed to narrow HandlerStatisticsDataReader\n")));
+    return;
+  }
+
+  HandlerStatisticsSeq data;
+  DDS::SampleInfoSeq infos;
+  DDS::ReturnCode_t ret = dr->read(data,
+                                   infos,
+                                   DDS::LENGTH_UNLIMITED,
+                                   DDS::NOT_READ_SAMPLE_STATE,
+                                   DDS::ANY_VIEW_STATE,
+                                   DDS::ALIVE_INSTANCE_STATE);
+  if (ret != DDS::RETCODE_OK) {
+    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) %N:%l ERROR: HandlerStatisticsListener::on_data_available failed to read\n")));
+    return;
+  }
+
+  ACE_TCHAR timestamp[27];
```

Comment:
Does ACE have a constant for this one?  If not we could add one in OpenDDS common libs?

Reply:
The code that I stole from didn't use one.  Where do you want it and what do you want it to be named?

Reply:
`dds/DCPS/Definitions.h` ACE_timestamp_length (or similar?)